aws-amplify/amplify-ui
# fix(react-storage): nested folder uploads

#### Description of changes
Fixes a bug where folder uploads got uploaded to the base location where upload was done rather than the folder structure path. 
We do so by using the `webKitRelativepath` value when available which gives us the complete folder structure path.



#### Issue #, if available



#### Description of how you validated changes
Manual tests and unit tests
- Upload a folder with nested structure.
- See that the uploaded files are in the same nested structure.

#### Checklist



- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
